src/Uno.UI.Runtime.Skia.Wpf/Extensions/Storage/Pickers/FolderPickerExtension.cs

```diff
@@ -0,0 +1,67 @@
+﻿#nullable enable
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Uno.UI.Runtime.Skia;
+using Windows.Storage;
+using Windows.Storage.Pickers;
+
+namespace Uno.Extensions.Storage.Pickers
+{
+	internal class FolderPickerExtension : IFolderPickerExtension
+	{
+		private readonly FolderPicker _picker;
+
+		public FolderPickerExtension(FolderPicker owner)
+		{
+			_picker = owner ?? throw new ArgumentNullException(nameof(owner));
+		}
+
+		public async Task<StorageFolder?> PickSingleFolderAsync(CancellationToken token)
+		{
+			NativeMethods.BROWSEINFO pbi = new NativeMethods.BROWSEINFO();
+			pbi.ulFlags = 0x41;
+			pbi.pszDisplayName = new string('\0', 256);
+
+			IntPtr pidl = NativeMethods.SHBrowseForFolder(ref pbi);
+			if (pidl != IntPtr.Zero)
+			{
+				IntPtr pathp = Marshal.AllocHGlobal(256);
+				bool success = NativeMethods.SHGetPathFromIDList(pidl, pathp);
+				Marshal.FreeCoTaskMem(pidl);
+
+				if (success)
+				{
+					string path = Marshal.PtrToStringAnsi(pathp);
+					Marshal.FreeHGlobal(pathp);
```

Comment:
Also move this to a finally block, as in this case, if `success` is false, it would be leaking the allocation.